src/create-template-orders.js

```diff
@@ -20,20 +33,263 @@ async function _fetchLastStartTimestamp() {
+      if (retryCount > maxRetry) {
+        const retryMessage =
+          'Got a concurrent modification error' +
+          ` when creating template order with number "${templateOrder.orderNumber}".` +
+          ` Version tried "${version}",` +
+          ` currentVersion: "${currentVersion}".`
+        throw new VError(
+          err,
+          `${retryMessage} Won't retry again` +
+            ` because of a reached limit ${maxRetry}` +
+            ' max retries.'
+        )
+      }
+      version = currentVersion
+    }
+}
+
+async function _createTemplateOrderAndPayments(checkoutOrder, orderDraft) {
+  try {
+    const checkoutPayments = checkoutOrder.paymentInfo?.payments?.map(
+      (p) => p.obj
+    )
+    let addPaymentActions
+    if (checkoutPayments) {
+      const paymentDrafts = checkoutPayments.map(_createPaymentDraft)
+      const paymentCreateResponses = await Promise.all(
+        paymentDrafts.map((paymentDraft) =>
+          apiRoot.payments().post({ body: paymentDraft }).execute()
+        )
+      )
+      addPaymentActions = paymentCreateResponses.map(
+        (paymentCreateResponse) => ({
+          action: 'addPayment',
+          payment: {
+            type: 'payment',
+            id: paymentCreateResponse.body.id,
+          },
+        })
+      )
+    }
+    const { body: templateOrder } = await apiRoot
+      .orders()
+      .importOrder()
+      .post({ body: orderDraft })
+      .execute()
+
+    if (addPaymentActions)
+      await _addPaymentWithRetry(templateOrder, addPaymentActions)
+  } catch (err) {
+    if (!isDuplicateOrderError(err)) {
+      const errMsg =
+        `Unexpected error on creating template order from checkout order with number: ${checkoutOrder.orderNumber}.` +
+        ` Line item: ${JSON.stringify(orderDraft.lineItems)}`
+      throw new VError(err, errMsg)
+    }
+  }
+}
+
+async function _setCheckoutOrderProcessed(checkoutOrder) {
+  const updateActions = [
+    {
+      action: 'setCustomField',
+      name: 'isSubscriptionProcessed',
+      value: true,
+    },
+  ]
+  await apiRoot
+    .orders()
+    .withId({ ID: checkoutOrder.id })
+    .post({
+      body: {
+        actions: updateActions,
+        version: checkoutOrder.version,
+      },
+    })
+    .execute()
+}
+
+async function _processCheckoutOrder(checkoutOrder) {
+  try {
+    const templateOrderDrafts = _generateTemplateOrderDrafts(checkoutOrder)
+
+    await pMap(
+      templateOrderDrafts,
+      async (templateOrderDraft) => {
+        await _createTemplateOrderAndPayments(checkoutOrder, templateOrderDraft)
+      },
+      { concurrency: 3 }
+    )
+
+    await _setCheckoutOrderProcessed(checkoutOrder)
+  } catch (err) {
+    logger.error(
+      `Failed to create template order from the checkout order with number ${checkoutOrder.orderNumber}. ` +
+        'Skipping this checkout order' +
+        ` Error: ${JSON.stringify(serializeError(err))}`
+    )
+  }
+}
+
+function isDuplicateOrderError(e) {
```

Comment:
I think this is a private function too, so you could use the convention _ for the function name prefix, I'm also thinking that maybe those conventions are not useful at all as we could use export if it's needs to be exported/shared. wdyt ?

Reply:
Indeed it's not needed. It's just a convention so when you read the code, you know immediately which function is private.